routing_model/src/flows_src.rs

```diff
@@ -184,10 +180,10 @@ impl StartRelocateSrc {
 
     fn allow_resend(&self, candidate: Candidate) -> Self {
         let mut state = self.clone();
-        state
+        unwrap!(state
```

Comment:
why unwrap in some cases and assert in others?

Reply:
`unwrap!` is for values that have `Some` or `Ok`.  We have `unwrap_err!` for `Err`, but there's no `unwrap!` equivalent for `None`, so I used `assert!` for those.

Reply:
They do seem to have a semantic difference, unwrap suggest we use the value, assert suggest we are testing an invariant/ pre/post-condition.